### ISSUE TITLE
Unsupported operation: Cannot modify unmodifiable map

### ISSUE THREAD

**Issue by skyrid3r698** [Teifun2/nextcloud-cookbook-flutter#186]
Recently, I cannot create new recipes with your mobile app on my Android Phones. I always get this error
![Screenshot_20230403_192806](https://user-images.githubusercontent.com/79767107/229583571-1a9cd500-a843-4665-81fa-1aad7a7ff1e9.jpg)

The Webapp still works perfectly fine.
Android App Version: 0.7.9
Webapp Version: 0.10.2
Nextcloud Version: 25.0.5

**Comment by Leptopoda** [comment]
Can you please add information on what you where trying to do when the error poped up?
Was it just hitting the submit button or when you tried adding ingredients?

I have an idea where this comes from but that information would make it easier finding.

**Comment by Leptopoda** [comment]
Ok I found the issue. #171 is not affected by this

**Comment by skyrid3r698** [comment]
Yes, as soon as I click "Create" this error pops up. #171 looks like a big leap and looks like it is about to be merged?

**Comment by Leptopoda** [comment]
Yep i'ts already merged and all api endpoints and data structure classes are now covered in unit tests.
The bug actually was when constructing a Recipe class.

Sadly it will still takes a few weeks for a new release as some behavior has to be re implemented.

**Comment by nicofrand** [comment]
Hi,

Did anyone find a workaround (until the new great release is out)? Would downgrading the version help (I could not via F-Droid though)?

**Comment by Creat** [comment]
Just to add my "+1" to this. I have freshly installed the cookbook on my nextcloud instance and this app on my phone. I can't create recipies at all on android.

What I did: Click the "+" on the main screen. Then just entering a `recipe name` (no other information) and clicking the `Create` button at the bottom throws the exception as OP has shown in his screenshot.

I am able to modify an existing recipe though.

**Comment by SK-Crow** [comment]
I am also having the same problem on the latest version.

**Comment by asandikci** [comment]
I have same problem too

**Comment by FritzThird** [comment]
I'm having the same problem.  I'm using NextCloud Cookbook version 1.1 for iOS.

![image](https://github.com/Teifun2/nextcloud-cookbook-flutter/assets/35336464/2fe2d4f0-2979-4e38-872e-88f9748fb700)

**Comment by nicofrand** [comment]
> Yep i'ts already merged and all api endpoints and data structure classes are now covered in unit tests. The bug actually was when constructing a Recipe class.
> 
> Sadly it will still takes a few weeks for a new release as some behavior has to be re implemented.

Hi! Is there a ticket or roadmap for the things that are still missing before being able to release a new version?
If possible I'd like to help.

**Comment by tinidisco2** [comment]
Hello, I have same problem too. I really like this cookbook app and it would be great if I could also create new recipes with my android phone. 
The temporary solution is to create a new recipe in the Nook app and just enter its name. It can then be edited in this application as well.

**Comment by Qfrijters** [comment]
just to bump it back up. I have this issue as well.

**Comment by suka97** [comment]
Same here, freshly installed both app and server

**Comment by gsalvatella** [comment]
Same problem here. This is definitely a showstopper. What needs to be addressed to fix it?
